### DISCUSSION TITLE
Error Websocket: Log Watcher: Server Connect Error: Error: websocket error (http://server ip:3012)

### DISCUSSION THREAD

**Discussion by andrec84j** [jhuckaby/Cronicle#525] [General]
I have a python script that makes a connection to a database that takes about 1 minute to be executed, at which time an error message appears in the event log:

Log Watcher: Server Connect Error: Error: websocket error (http://server ip:3012) * 

How can I solve this problem?

Target called to own server

Thanks

**Comment by jhuckaby** [comment]
The Cronicle UI makes a direct websocket connection to the worker server running the log, to stream the live log to the client (user).  It looks like your worker server isn't accessible via its LAN IP on the Cronicle port (3012), so the live log cannot be displayed.  There are a wide variety of reasons why this would be the case, all depending on your server architecture.  Play with some of these configuration options to see what works for you:

- https://github.com/jhuckaby/Cronicle/blob/master/docs/Configuration.md#web_direct_connect
- https://github.com/jhuckaby/Cronicle/blob/master/docs/Configuration.md#web_socket_use_hostnames
- https://github.com/jhuckaby/Cronicle/blob/master/docs/Configuration.md#socket_io_transports

Note that Cronicle is still working, and your job still completes (and the job output should appear at that point).  This is just an issue with displaying the live log in real-time.

Hope this helps.

**Reply by Drallas** [reply]
I'm having this issue behind my Nginx Proxy Manager. 

Websockets is on, without it get a partly blank cronicle screen. 

Logging in via ip:port I do see the live logs, but not via mydomein.ext!?

I played around with the recommended settings above, but unfortunately without any luck!

**Comment by mikeTWC1984** [comment]
@Drallas if you are using reverse proxy it might get complicated with ws. Check your browser console for errors. One of the issues I recall is that you need to provide extra route for socket.io in you nginx config. Take a look:
https://github.com/cronicle-edge/cronicle-edge/blob/main/Docker/nginx.conf

Also, if your proxy upgrade http to https then websocket will not work (by design)

**Reply by Drallas** [reply]
I guess I sit it out until v 2.0 using http://ip:3012 when I want to see live logs.

**Comment by mikeTWC1984** [comment]
Similar issue was just discussed
https://github.com/jhuckaby/Cronicle/discussions/644

@jhuckaby  are you going to use built-in browser ws-client on the 2.0? On of the thing causing issue like above is that by default socket.io serves it's client library automatically (it's not loaded from htdocs), so that link is lost behind proxy.

**Comment by jhuckaby** [comment]
Yup, v2.0 uses native [WebSockets](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket) in the browser (and the amazing [ws](https://www.npmjs.com/package/ws) module on the server-side).  This will simplify everything greatly, and eliminate many problems people are having with socket.io and proxies.

Also with v2.0, everything (and I mean *everything*) goes through a *single* native WebSocket connection to the master server.  This should fix a ton of problems people are having with live log view, etc.

Also, HTTPS (SSL) is 100% fully supported end-to-end, and it is now the default.
